src/kernel/src/device.rs

```diff
@@ -65,11 +67,47 @@ fn get_kso_manager() -> &'static KsoManager {
     })
 }
 
-pub fn kaction(cmd: KactionCmd, id: Option<ObjID>, arg: u64) -> Result<KactionValue, KactionError> {
+pub fn kaction(
+    cmd: KactionCmd,
+    id: Option<ObjID>,
+    arg: u64,
+    arg2: u64,
+) -> Result<KactionValue, KactionError> {
     match cmd {
         KactionCmd::Generic(cmd) => match cmd {
-            KactionGenericCmd::AllocateDMA(_np) => {
-                todo!()
+            KactionGenericCmd::ReleasePin => {
+                let id = id.ok_or(KactionError::InvalidArgument)?;
+                let obj = lookup_object(id, LookupFlags::empty()).ok_or(KactionError::NotFound)?;
+                let pin = arg as u32;
+                obj.release_pin(pin);
+                Ok(KactionValue::U64(0))
+            }
+            KactionGenericCmd::PinPages(_np) => {
+                let id = id.ok_or(KactionError::InvalidArgument)?;
+                let obj = lookup_object(id, LookupFlags::empty()).ok_or(KactionError::NotFound)?;
+
+                let start = (arg2 & 0xffffffff) as usize;
```

Comment:
If both these values, 0xffffffff and 32 are going to be commonly used can you please name them rather than using magic numbers?

Reply:
Sure